modules/blocks/chain.js

```diff
@@ -736,37 +739,49 @@ Chain.prototype.deleteLastBlock = function(cb) {
 		return setImmediate(cb, 'Cannot delete genesis block');
 	}
 
-	async.waterfall(
-		[
-			waterCb => {
-				// Delete last block, replace last block with previous block, undo things
-				__private.popLastBlock(lastBlock, (err, newLastBlock) => {
+	let deletedBlockTransactions;
+
+	async.series(
+		{
+			popLastBlock(seriesCb) {
+				// Perform actual delete of last block
+				__private.popLastBlock(lastBlock, (err, previousBlock) => {
```

Comment:
Follow the convention that we agreed upon. and use `popLastBlockErr` instead of `err`